src/cpp/scaler/ymq/internal/raw_stream_client_handle.h

```diff
@@ -10,6 +10,7 @@ namespace ymq {
 class RawStreamClientHandle {
 public:
     RawStreamClientHandle(sockaddr remoteAddr);
+    RawStreamClientHandle(sockaddr_un remoteAddr);
```

Comment:
I think better way to solve this would be for this API to not expose the system address types (`sockaddr` and `sockaddr_un`) but a custom address type. 

```suggestion
    RawStreamClientHandle(TCPAddress remoteAddr);
    RawStreamClientHandle(IPCAddress remoteAddr);
```

1. that would gracefully solve the fallback definition of `sockaddr_un`, and the not ideal `_addrSize` logic for checking if an address is TCP or IPC;
2. the parsing logic for these two types of addresses could be entirely isolated from `IOSocket`.